ext/standard/config.m4

```diff
@@ -259,24 +268,23 @@ int main() {
 	return 1;
 #endif
 }]])],[
-  ac_cv_crypt_sha256=yes
-],[
-  ac_cv_crypt_sha256=no
-],[
-  ac_cv_crypt_sha256=no
-])])
+    ac_cv_crypt_sha256=yes
+  ],[
+    ac_cv_crypt_sha256=no
+  ],[
+    ac_cv_crypt_sha256=no
+  ])])
 
 
-dnl
-dnl If one of them is missing, use our own implementation, portable code is then possible
-dnl
-dnl TODO This is currently always enabled
-if test "$ac_cv_crypt_blowfish" = "no" || test "$ac_cv_crypt_des" = "no" || test "$ac_cv_crypt_ext_des" = "no" || test "$ac_cv_crypt_md5" = "no" || test "$ac_cv_crypt_sha512" = "no" || test "$ac_cv_crypt_sha256" = "no" || test "$ac_cv_func_crypt_r" != "yes" || true; then
+  if test "$ac_cv_crypt_blowfish" = "no" || test "$ac_cv_crypt_des" = "no" || test "$ac_cv_crypt_ext_des" = "no" || test "$ac_cv_crypt_md5" = "no" || test "$ac_cv_crypt_sha512" = "no" || test "$ac_cv_crypt_sha256" = "no" || test "$ac_cv_func_crypt_r" != "yes"; then
+    AC_MSG_ERROR([Cannot use external libcrypt as some algo are missing])
+  fi
+
+  AC_DEFINE_UNQUOTED(PHP_USE_PHP_CRYPT_R, 0, [Whether PHP has to use its own crypt_r for blowfish, des and ext des])
```

Comment:
I'd drop the "for blowfish ..." part at the end -- it's different between the two messages and the list is incomplete anyway.